debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/DateTimeFormat.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.postgresql.connection.wal2json;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+import java.util.function.Supplier;
+
+import org.apache.kafka.connect.errors.ConnectException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.time.NanoTimestamp;
+
+/**
+ * Transformer for time/date related string representation in JSON message coming from plugin.
+ * 
+ * @author Jiri Pechanec
+ *
+ */
+public interface DateTimeFormat {
+    public long timestamp(final String s);
+    public long timestampWithTimeZone(final String s);
+    public long systemTimestamp(final String s);
+    public LocalDate date(final String s);
+    public LocalTime time(final String s);
+    public OffsetTime timeWithTimeZone(final String s);
+
+    public static DateTimeFormat get() {
+        return new ISODateTimeFormat();
+    }
+    public static class ISODateTimeFormat implements DateTimeFormat {
+        private final Logger logger = LoggerFactory.getLogger(getClass());
+
+        private static final String TS_FORMAT_PATTERN = "yyyy-MM-dd HH:mm:ss";
+        private static final DateTimeFormatter TS_FORMAT = DateTimeFormatter.ofPattern(TS_FORMAT_PATTERN);
+
+        private static final String TS_TZ_FORMAT_PATTERN = "yyyy-MM-dd HH:mm:ssX";
+        private static final DateTimeFormatter TS_TZ_FORMAT = DateTimeFormatter.ofPattern(TS_TZ_FORMAT_PATTERN);
```

Comment:
Hi all. I was playing with the branch and caught an error: 
Cannot parse time/date value '2017-10-05 05:57:13.4+00', expected format 'yyyy-MM-dd HH:mm:ssX'   
Is it known limitation on timestamp format or a bug in the DateTimeFormat ?

Reply:
Awesome, thanks for the feedback! Is it caused by datatype TIMESTAMP(1) WITH TZ? If yes, than the fix should be pretty easy.

Thanks

Reply:
I'm afraid it will be not so easy. The column type of all my date columns is just TIMESTAMPTZ. I've updated my table again and checked replication slot value via SELECT * FROM pg_logical_slot_get_changes (see attachment); And it seems, that wal2json plugin cuts of ending zeros from timestemps.
[event.txt](https://github.com/debezium/debezium/files/1367750/event.txt)

Reply:
I can change the format into `yyyy-MM-dd HH:mm:ss[.S]X which should make an fraction second an optional section and use `parseBest` instead of `parse`.

Reply:
Yes, but the root cause of the issue is that wal2json output plugin produces different number of fraction seconds (as showed in attachment), despite the fact, that column types is just TIMEZTAMPTZ
db data: "2017-07-13 17:03:59.544000+03" -> slot data: "2017-07-13 17:03:59.544+03",
db data: "2017-10-09 10:16:39.712883+03" -> slot data: "2017-10-09 10:16:39.712883+03",
db data: "2017-10-05 08:57:13.400000+03" -> slot data: "2017-10-05 08:57:13.4+03"

Reply:
Yes, I understand but we are not in control of wal2json plug-in so we must be prepared to handle such imperefect input data.